### DISCUSSION TITLE
docker_base_image is called even if running_within_docker false

### DISCUSSION THREAD

**Discussion by tessus** [dani-garcia/vaultwarden#3021] [Ideas]
I've noticed that the support string still shows and calls `docker_base_image()`, even though `running_within_docker` is false.

The checks on the diagnostics show:

```
Running within Docker         No 
```

but the support string shows:

```
* Running within Docker: false (Base: Unknown)
```

I'd like to propose either one of the 2 options (I'll create a PR, if and when there's a consensus):

- do not show `(Base: Unknown)`, if `Running within Docker: false`
- read `/etc/os-release`, if it exists and use `NAME` or `PRETTY_NAME` as base (apart from Debian and Alpine)

@BlackDex what's your take on this? Does this make sense or am I overthinking it?

**Comment by BlackDex** [comment]
While that isn't that big of an issue i think. I have thought about this. Adding the complexity of detected the OS of the user.
But i think that isn't something we need. For me, it only is useful to know which base OS is used for the images we build.

That call isn't that big of an issue, doesn't take much resources. I'm currently working on some other changes on the diagnostics page, i can take this change with it by not calling it when not in docker.

**Reply by tessus** [reply]
Thanks for the reply. It certainly isn't important, rather a cosmetic issue, but it would have given me the opportunity to dabble a bit more in the Rust lang.

**Reply by BlackDex** [reply]
Your always allowed to dabble with it. I'm not sure if there is something on the meta list which is easy to pickup and not yet there.

**Reply by BlackDex** [reply]
Besides that, you of course can create a pr which puts an if..else around it. But i doubt that was the dabble you were looking for.
